### PR TITLE
Version ticks for 3.10.2 stable release

### PR DESCRIPTION
**Pull request by AustinSanders**
Ticked the version numbers and set variables for 3.10.2 release.

## Description
Ticked the version numbers and set variables for 3.10.2 release. Copied contents of environment.yaml to meta build/run sections.

## Related Issue
https://trello.com/c/IUrphCSv/362-do-the-release

## Motivation and Context
Readies the package for release.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -107,13 +104,11 @@ requirements:
   run:
   - armadillo==8.200.0
   - blas==1.1
-  - boost==1.67
```
Comment (jessemapel) [inline recipe/meta.yaml]: Why is this removed? Xalan and Doxygen are not needed because they are for building docs.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -129,15 +124,15 @@ requirements:
...
+  - libtiff>=4.0.9
   - libxml2==2.9.7
   - mesalib==17.2.0
   - mysql==5.7.20
   - mysql-connector-c==6.1.6
   - nanoflann==1.2.2
   - conda-forge/label/gcc7::nn
   - openblas==0.2.19
-  - opencv==3.2.0
+  - opencv
```
Comment (jessemapel) [inline recipe/meta.yaml]: Opening up OpenCV can break a lot of things. We have this hard pinned to 3.2.0 so that we get the "non-free" packages and the correct interface for findfeatures. Why was this changed?

**Review by jessemapel — COMMENTED** [review]
Version # and CMake changes look good. Why was the environment changed?

File: recipe/meta.yaml (hunk 1 above)
Comment (AustinSanders) [reply recipe/meta.yaml]: I copied the dependencies directly from the environment.yml dependencies section.  Boost was not in environment.yml.

File: recipe/meta.yaml (hunk 2 above)
Comment (AustinSanders) [reply recipe/meta.yaml]: The section was copied from environment.yml, which does not specify a version for opencv.

Comment (AustinSanders) [reply recipe/meta.yaml]: I'll change environment back to including boost + explicitly versioning opencv, but should I also change the environment.yml file?  Or change the docs that state that these sections should be copied from environment.yml?  Things seem a little out of sync here.

Comment (jessemapel) [reply recipe/meta.yaml]: The two files are out of synce because of how channels are specified in the two. In an environment file we can specify a channel priority, in a conda-build meta.yml file we cannot. Having things more pinned in the meta.yaml helps ensure all of the correct packages are installed without the channel priority. For the 4.1 release, modifying the meta.yaml make sense as we are working on dependencies, but the 3.10 release doesn't have any of that in it, so we want to keep using the old environment.

**Review by jessemapel — APPROVED** [review]
